anxcloud/struct_dns_zones_test.go

```diff
@@ -0,0 +1,120 @@
+package anxcloud
+
+import (
+	"github.com/anexia-it/go-anxcloud/pkg/clouddns/zone"
+	"github.com/google/go-cmp/cmp"
+	"testing"
+)
+
+func TestFlattenDnsZones(t *testing.T) {
+	cases := []struct {
+		Input          []zone.Zone
+		ExpectedOutput []interface{}
+	}{
+		{
+			[]zone.Zone{
+				{
+					Definition: &zone.Definition{
+						Name:             "zone1.test",
+						IsMaster:         true,
+						DNSSecMode:       "unvalidated",
+						AdminEmail:       "test@zone1.test",
+						Refresh:          3600,
+						Retry:            300,
+						Expire:           3600,
+						TTL:              60,
+						NotifyAllowedIPs: []string{"127.0.0.1", "192.168.0.1"},
+						MasterNS:         "8.8.8.8",
+						DNSServers: []zone.DNSServer{
+							{
+								Server: "nameserver-1",
+								Alias:  "ns1",
+							},
+						},
+					},
+					DeploymentLevel: 100,
+					ValidationLevel: 100,
+					IsEditable:      true,
+				},
+				{
+					Definition: &zone.Definition{
+						Name:             "zone2.test",
+						IsMaster:         true,
+						DNSSecMode:       "managed",
+						AdminEmail:       "test@zone2.test",
+						Refresh:          3600,
+						Retry:            300,
+						Expire:           3600,
+						TTL:              60,
+						NotifyAllowedIPs: []string{"127.0.0.1", "192.168.0.1"},
+						MasterNS:         "8.8.8.8",
+						DNSServers: []zone.DNSServer{
+							{
+								Server: "nameserver-2",
+								Alias:  "ns2",
+							},
+						},
+					},
+					DeploymentLevel: 99,
+					ValidationLevel: 99,
+					IsEditable:      false,
+				},
+			},
+			[]interface{}{
+				map[string]interface{}{
```

Comment:
hm when we are using this type outside of a single function, we should really name it ..

Reply:
scrap that, consistency is more important than this one

Reply:
Due to consistnecy I would like to keep it this way.